torch/_refs/_conversions.py

```diff
@@ -104,3 +105,14 @@ def complex(real: TensorLikeType, imag: TensorLikeType) -> TensorLikeType:
     result.real = real
     result.imag = imag
     return result
+
+
+@register_decomposition(torch._ops.ops.aten.polar)
+# Note: polar has type promotion tests disabled due to different semantics.
+# exact_dtype is for compat with complex_check_dtype from core.
+@out_wrapper(exact_dtype=True)
+def polar(abs: TensorLikeType, angle: TensorLikeType) -> TensorLikeType:
+    result = torch.complex(abs, angle)
+    result.real = abs * torch.cos(angle)
+    result.imag = abs * torch.sin(angle)
```

Comment:
This is... not a very good ref lol

Reply:
What exactly is the problem? The CPU kernel is defined as:
```
c10::complex<scalar_t>(a * std::cos(b), a * std::sin(b))
```

Reply:
The main problem is it is implemented with mutation which makes it compiler unfriendly. The complex construction is goofy but subsidiary to this

Reply:
OK. I'm going to merge it anyway, though, since it fixes some test failures.